subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/MinimalJavaCompileOptions.java

```diff
@@ -25,13 +25,15 @@
 import javax.annotation.Nullable;
 import java.io.File;
 import java.io.Serializable;
+import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 
 public class MinimalJavaCompileOptions implements Serializable {
     private List<File> sourcepath;
     private List<String> compilerArgs;
     private String encoding;
-    private String bootClasspath;
+    private Set<File> bootClasspath;
```

Comment:
I don't think this should be a `Set` (or a `List`) at this point. If we already do a breaking change, I would change `FileCollection.getAsPath()` to do the validation.

Even better, we could emit a deprecation warning there right now and fail in Gradle 8.0. WDYT?

Reply:
It would definitely work also with other cases. I will change it.

Reply:
fixed